sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/CustomEvents/IApplicationInsightsEventLogger.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+
+namespace Azure.Monitor.OpenTelemetry.Events
+{
+    /// <summary>
+    /// An interface for logging custom events telemetry.
+    /// </summary>
+    public interface IApplicationInsightsEventLogger
+    {
+        /// <summary>
+        /// Tracks Application Insights custom events.
+        /// </summary>
+        /// <param name="name">Name of the CustomEvent.</param>
+        /// <param name="attributes">Custom dimensions of the event.</param>
+        public void TrackEvent(string name, IReadOnlyList<KeyValuePair<string, object?>>? attributes = null);
+    }
```

Comment:
Some other possible shapes to consider:

```csharp
public void TrackEvent(string name, in TagList attributes = default);
public void TrackEvent(string name, params ReadOnlySpan<KeyValuePair<string, object?>> attributes); // May be hard/impossible to implement with ILogger or require newer runtime features but would be awesome to support :)
```

Reply:
Like the Taglist option. @rajkumar-rangaraj - let me know what you think, can change it as a follow up.